File: 부품 찾기.py
def binary_search(array, target, start, end):
	if start > end:
		return None
	mid = (start + end) // 2
	if array[mid] == target:
		return mid # 리스트의 인덱스를 반환
	elif array[mid] > target: # 찾는 값이 중간 값보다 작은 경우
		return binary_search(array, target, start, mid -1) # 끝점을 중간 -1로 잡고, 다시 수행
	else :                    # 찾는 값이 중간 값보다 큰 경우
		return binary_search(array, target, mid+1, end)  # 시작 점을 중간 + 1로 잡고, 다시 수행
n = int(input())
# target = list(map(int,input().split())) #원소의 개수, 찾고자 하는 문자 입력
array = list(map(int,input().split()))
array.sort()
m = int(input())
x = list(map(int,input().split()))
#이진 탐색 수행
for i in x:
  result = binary_search(array,i,0,n-1)
  if result != None:
    print("yes",end = ' ')
  else:
    print("no", end = ' ')
    
   

File: 부품 찾기.java
import java.util.*;

public class Main{
  public static int binary_search(int[] array,int target,int start,int end){
    if(start>end){
      return -1;
    }
    int mid = (start+end) / 2;
    if(target == array[mid]){
      return mid;
    }
    else if(target < array[mid]){
      return binary_search(array, target, start, mid-1);
    }
    else{
      return binary_search(array, target, mid+1, end);
    }
  }
  public static void main(String[] args){
    Scanner sc = new Scanner(System.in);

    int n = sc.nextInt();
    int[] arr = new int[n];
    for(int i=0;i<n;i++){
      arr[i] = sc.nextInt();
    }

    int target = sc.nextInt();
    int[] arr2 = new int[target];
    for(int i =0;i<target;i++){
      arr2[i] = sc.nextInt();
    }
    Arrays.sort(arr);
    
    for(int i =0;i<target;i++){
      int result = binary_search(arr,arr2[i],0,n);
      if(result == -1){
        System.out.print("no ");
      }
      else{
        System.out.print("yes ");
      }
    }
  
  }
}
